rest/docs/design/technical_design.md

```diff
@@ -0,0 +1,127 @@
+Steps are declared within an `Integration`.
+
+
+The overall idea is that your `Integration` define 2 kind of information:
+1. a catalog of technologies and allowed operations (`Connectors`,  `Actions`)
+2. a pipeline of customized components(`Connection`) and operations you need to perform(`Step`), using the building blocks defined at 1).
+
+To reinforce the idea you can see it as if:
+
+You define some (technology)`Connector`, where you declare all the properties you allow the developer to give a value to in a corresponding set of `Connection`s. You also link this `Connector` with a series of operations that are tied to this technology, represented by `Action`. In each `Action` you declare also eventual additional properties.  
+At last, you detail how the data will flow across `Connections`, and with which eventual tranformation, specifying a list of `Steps`.
+
+Notice that despite you define an ordered list of `Steps` to define the business behavior of your `Integration` there is no type checking of any sort, between those steps. It's up to you be sure that the output of a `Step` is coherent with the output of the following one.
+
+
+## Integration creation workflow
+
+The entities described and linked above are indeed what the code implementation will use to implement the required behavior. But the idea is that those elements are protected behind some layer of abstraction that allows you not to manipulate them directly.
+
+From a UX point of view, the user is exposed to a graphical UI, dynamically built base on the above models.
+Interacting with a set of forms and UI elements, a user completes the definition of a logical integration.
+
+When she's done, the UI application produces a `.json` files that corresponds to the user activity and that will be used as the input to generate code on his behalf.
+
+The component responsible of consuming this `.json` is the [`ActivateHandler`](https://github.com/syndesisio/syndesis/blob/master/rest/controllers/src/main/java/io/syndesis/controllers/integration/online/ActivateHandler.java), that at runtime, lives in the `syndesis-rest` pod.  
+
+This component is responsible of 3 main activities:  
+1. produce **deplyoment data** 
+1. **build** the project related to our `Integration`
+1. **deploy** an instance of our `Integration`
+
+
+
+##### Build
+**Building** is actually delegated to an implementation of [`ProjectGenerator`](https://github.com/syndesisio/syndesis/blob/master/rest/project-generator/src/main/java/io/syndesis/project/converter/DefaultProjectGenerator.java) that is responsible of **build time code generation**.  
+With some introspection and templating, a full Java based Apache Maven project is generated. The idea is that this artifact could be built succesfully even outside Syndesis runtime platform, being a fully consistent project.  
```

Comment:
Is this project accessible to users?
Is it possible to them to extract it and run it locally for investigation?
If yes, it would be nice that Fuse Tooling can support this kind of projects.